Documentation/Storage-Configuration/Advanced/key-management-system.md

```diff
@@ -334,3 +338,45 @@ security:
     # name of the k8s secret containing the credentials.
     tokenSecretName: kmip-credentials
 ```
+
+## Azure Key Vault
+
+Rook supports storing OSD encryption in [Azure Key vault](https://learn.microsoft.com/en-us/azure/key-vault/general/quick-create-portal)
+
+### Authentication Methods
+
+Rook support [Authentication with Service Principal](https://learn.microsoft.com/en-us/azure/key-vault/general/authentication) in order to connect with the Azure Key Vault. The service principal can be created either using a secret or a certificate.
+
+#### Authenticate Service Principal with a Secret
+
+Rook operator can authenticate with the Azure service principal using the client sercert. In this case the cephCluster spec will look like:
+
+```yaml
+security:
+  kms:
+    connectionDetails:
+      KMS_PROVIDER: azure
+      AZURE_VAULT_ADDR: https://<key-vault name>.vault.azure.net/0
+      AZURE_CLIENT_ID: Application ID of an Azure service principal
+      AZURE_TENANT_ID: ID of the application's Microsoft Entra tenant
+      AZURE_CLIENT_SECRET: <name of the k8s secret containing Password of the Azure service principal>
+```
+
+Check the [azure documentation](https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication-service-principal?tabs=azure-cli#-option-1-create-an-azure-service-principal-with-a-secret) for creating a azure service principal with a secret.
+
+#### Authenticate Service Principal with a certificate
+
+This is an advanced but recommended configuration for production deployments. In this case the cephCluster spec will look like:
+
+```yaml
+security:
+  kms:
+    connectionDetails:
+      KMS_PROVIDER: azure
+      AZURE_VAULT_ADDR: https://<key-vault name>.vault.azure.net/0
+      AZURE_CLIENT_ID: Application ID of an Azure service principal
+      AZURE_TENANT_ID: ID of the application's Microsoft Entra tenant
+      AZURE_CLIENT_CERT: <name of the k8s secret containing the certificate along with the private key (without password protection)>
```

Comment:
We may need to tell users what fields need to be set in the Secret specifically. A quick look at the doc linked below doesn't suggest to me what the secret data key should be.

Reply:
@BlaineEXE I've tried to add more details and referred to more azure documentation. Please take a look again.